Merge latest csstudio

Tom-Willemsen opened the pull request:
Merge latest CSStudio into our master

Gets us the change that @DominicOram  made to serve JSON in archive engine as well as other improvements.

DominicOram commented:
The CommandsServlet uses the Oak Ridge copyright info, this is probably wrong, what should be used?

DominicOram commented:
After starting on ticket 971 it appears that more information will have to be provided by the server. The ticket can still be reviewed to test that some information is being sent but it should not yet be closed.

KathrynBaker commented:
I can certainly confirm that some information is provided by the server

Commit:
o.c.scan.diirt.datasource: add submit

Commit:
o.c.alarm.beast.configtool: add option to send JMS CONFIG message

DominicOram committed:
Added JSONWriter

DominicOram committed:
Made HTML responses more general

DominicOram committed:
Added JSON version of channel page

DominicOram committed:
Simplified JSONWriter

DominicOram committed:
Refactored response factory

DominicOram committed:
Added JSON for main page

DominicOram committed:
Added more docs

Commit:
Service button dialog layout fix. 

This fixes the layout of the Service method definition dialog on 4k
displays. A small layout fix is also made to Log entry dialog.

DominicOram committed:
Added JSON for group page

DominicOram committed:
Added JSON for disconnected and groups responses

DominicOram committed:
Merge branch 'master' into archive_engine_return_json_2265

berryma4 committed:
Merge pull request #2283 from css-iter/issue_2268-1

Service button dialog layout fix.

willrogers committed:
Prevent changing DB pv name to an empty string.

This could get the plot into an inconsistent state where you cannot
delete the item.

willrogers committed:
Databrowser autoscale to use 90% of available space.

On the vertical axis, this makes things feel a bit less cramped.

berryma4 committed:
Merge pull request #2286 from dls-controls/2284-db-delete-trace

Prevent changing DB pv name to an empty string.

berryma4 committed:
Merge pull request #2287 from dls-controls/2285-db-autoscale

Databrowser autoscale to use 90% of available space.

kasemir committed:
BOY PV Info: Show control low..high

kasemir committed:
BOY TextInput: Use PV _control_ limits for "Limits from PV"

.. instead of the display range.

#1910

DominicOram committed:
Changed JSON to use StringVTypeFormat

berryma4 committed:
Merge pull request #2288 from ControlSystemStudio/boy_pv_control_limit_1910

Boy Text Input: Use control instead of display range for "Limits from PV"

DominicOram committed:
Escaped characters in JSON

DominicOram committed:
Reverted accidental changes

Commit:
Fix for connection paint loop.

DominicOram committed:
Restructured to make more robust

berryma4 committed:
Merge pull request #2293 from css-iter/issue_2291

Fix for connection paint loop.

willrogers committed:
Avoid setDirty from non-UI thread in databrowser.

willrogers committed:
Databrowser: Fix invalid thread access in sample view

Commit:
@2x icons created.

Icons below listed  were resized from 16x16. They will have to be
replaced  with 32x32 sources, to get crisp look.
Icons from 16x16:
convertToEmbedded,copy_properties,distribute*,exeJS,ExitCompactMode,exitfullscen,js,jsembed
openfile,order,overview,paste_propeties,pyEmbedde,pythonFile,replace,reroute,rotate
setting,runtime,shape_mode,show,index,snap2geometry,sound,writePV,hideindex

berryma4 committed:
Merge pull request #2295 from dls-controls/2294-db-exceptions

Avoid setDirty from non-UI thread in databrowser.

berryma4 committed:
Merge pull request #2296 from dls-controls/2292-db-exceptions

Databrowser: Fix invalid thread access in sample view

Commit:
Additional @2x icons

DominicOram committed:
Added better checks for closed JSON structures and more tests.

DominicOram committed:
Channel values are now returned as a JSON object

Commit:
Fix for the multistate symbol memory leak error.

Fixed by rounding double values to long.

berryma4 committed:
Merge pull request #2297 from css-iter/issue_2268-2

@2x icons created.

nickbattam committed:
Move xmlunit from Package to Bundle requirement

berryma4 committed:
Merge pull request #2301 from dls-controls/xmlunit-depend

Move xmlunit from Package to Bundle requirement

nickbattam committed:
Replace spinner limits with .DRVH and .DRVL

berryma4 committed:
Merge pull request #2306 from ISISComputingGroup/archive_engine_return_json_2265

Modified archive engine to also serve JSON

nickbattam committed:
Add distinction between Control and Monitor PVs in widgets

berryma4 committed:
Merge pull request #2307 from dls-controls/2305-drivelimits

Respect DRVH/L in Knob and Slider widgets

willrogers committed:
Update pbrawclient jar to 0.0.8.

berryma4 committed:
Merge pull request #2304 from dls-controls/spinner-bounds

Replace spinner limits with .DRVH and .DRVL

willrogers committed:
Properly plumb in new AA client jar file.

willrogers committed:
Remove unused import.

nickbattam committed:
Update pbrawclient reference in build.properties

berryma4 committed:
Merge pull request #2310 from dls-controls/update-aa-pb-jar

2309 update aa pb jar

nickbattam committed:
Update to support getData for retired PVs

k-vijayan committed:
To retrieve retired PV from AA

nickbattam committed:
Update comment to explain argument

Commit:
Fix for the mixed arguments/results table

berryma4 committed:
Merge pull request #2313 from dls-controls/aa-retired-pv

Update to support getData for retired PVs

Commit:
AbsoluteTimeParser update with ISO time-stamp parsing. 

AbsoluteTimeParser -- class was updated with DateTimeForramter, which is
now used for parsing the input time-stamps. Parser also excepts full ISO
time-stamp formatting.

TimeParserUnitTest -- unit tests were added to support the full ISO
time-stamp in absolute time parser.

berryma4 committed:
Merge pull request #2315 from css-iter/issue_2268-3

Fix for the mixed arguments/results table in Service button dialog

berryma4 committed:
Merge pull request #2316 from css-iter/issue_2232-1

AbsoluteTimeParser update with ISO time-stamp parsing.

kasemir committed:
XYGraph: Only disable 'scrolling' for zoom/pan in time-based plot

#2318

berryma4 committed:
Merge pull request #2319 from ControlSystemStudio/xygraph_updates_stop_2318

XYGraph: Only disable 'scrolling' for zoom/pan in time-based plot

Commit:
Throttle change events from Scaled Slider

Commit:
Remove debug print statement

Commit:
Add comment explaining when listeners are called

berryma4 committed:
Merge pull request #2321 from css-iter/issue_2014

Fix for the multistate symbol memory leak error.

willrogers committed:
Update comments on scaled slider throttling.

berryma4 committed:
Merge pull request #2323 from dls-controls/issue_2320

Throttle the rate of updates when using the scaled slider widget

willrogers committed:
Default to PV limits on slider widgets.

If no limits are specified, default to the ones coming from the PV.

willrogers committed:
Default to precision from PV in text widgets.

If no precision is specified in EDM, default to getting it from the PV.

Commit:
Fix for no connections after restart 

Instance of SchemaService without dialog is created from first instance
of WidgetNodeEditPolicy. The dialog was interfering with the
connections.

berryma4 committed:
Merge pull request #2327 from dls-controls/2324-edm-converter

2324 edm converter

berryma4 committed:
Merge pull request #2328 from css-iter/issue_2326

Fix for no connections after restart

willrogers committed:
Don't use WorkbenchAdvisor.eventLoopIdle() to do background work.

The openDisplay jobs are being handled on background threads using asyncExec,
and the eventLoopIdle() method is not reliably called.

berryma4 committed:
Merge pull request #2330 from dls-controls/2329-launch-queue

Don't use WorkbenchAdvisor.eventLoopIdle() to do background work.

willrogers committed:
Copy StringSplitter.java to org.csstudio.opibuilder.converter.

This breaks the golden rule of avoiding code duplication, but it allows
building the EDM converter into a stand-alone jar file for external use. This
had previously been possible by including the org.csstudio.java plugin, but
since that developed a dependency on an Eclipse plugin that mechanism for
building proved to be tricky as well.

willrogers committed:
OPIShellSummary view: set minimum size on buttons.

Quite often the view is shrunk so you can't read the text on the buttons.

nickbattam committed:
Replace repeated hard-coded sizes with constants

berryma4 committed:
Merge pull request #2331 from dls-controls/2325-string-splitter

Copy StringSplitter.java to org.csstudio.opibuilder.converter.

nickbattam committed:
Remove trailing whitespace on empty line

berryma4 committed:
Merge pull request #2333 from dls-controls/2332-opi-summary-buttons

2332 opi summary buttons

willrogers committed:
Make bool button use cross cursor if not enabled.

willrogers committed:
Improve cursors on ScaledSlider.

If not in run mode, don't enable moving the slider and don't give visual
feedback. If in run mode but the slider is not enabled, show cross cursor
instead of hand cursor.

willrogers committed:
Allow scaled slider to be dragged in edit mode.

willrogers committed:
Replace scaled slider repaint() to its previous position.

kasemir committed:
TextInput: Only enforce min, max when min < max

Another attempt to fix the unfortunate client-side min/max enforcement
#1910

berryma4 committed:
Merge pull request #2335 from ControlSystemStudio/text_input_min_max_1910

TextInput: Only enforce min, max when min < max

willrogers committed:
Correct typo in prompt message.

berryma4 committed:
Merge pull request #2337 from ControlSystemStudio/issues_2170_master

o.c.scan.diirt.datasource: add submit (master - 4.5.x)

berryma4 committed:
Merge pull request #2338 from ControlSystemStudio/issue_2173_master

o.c.alarm.beast.configtool: add option to send JMS CONFIG message  (master - 4.5.x)

kasemir committed:
Include ADL (MEDM) converter

kasemir committed:
Revert "Include ADL (MEDM) converter"

This reverts commit 17fae19645c1e40886141e24e4299ff247eff714.

kasemir committed:
Start -application org.csstudio.opibuilder.adl2boy.application

kasemir committed:
MEDM converter: Handle files on command line

kasemir committed:
MEDM converter: Tweak printouts

kasemir committed:
MEDM cmd line converter: Javadoc

Commit:
Bug fix for issue 2345.

There was no content of the image on first screenshot instance.
Workaround is calling the gc.copyArea(image, 0, 0) twice.

Commit:
New message history OPI widget

Changes made also to the existing message history:
-Added end time to preference
-Query filtering that parses SQL like syntax and sets the corresponding
filters
-Added time format preference
-Ability to create new instances of message history widget
-Column filtering
-Renamed property from max properties to max messages.
-Changed the SQL for max messages, so it limits the actual messages.

berryma4 committed:
Merge pull request #2343 from ControlSystemStudio/MEDM_convert_2342

Medm convert 2342

kasemir committed:
BOY: Be tolerant to font and color.def with display builder additions

#2347

berryma4 committed:
Merge pull request #2348 from ControlSystemStudio/BOY_fonts_colors_2347

BOY: Be tolerant to font and color.def with display builder additions

berryma4 committed:
Merge pull request #2346 from css-iter/issue_2345

Bug fix for issue 2345.

Commit:
Added session persistence and filter display to the Message history
widget

Commit:
With the new JDOM 1.1.3 in maven-osgi-bundles the XInclude feature in OPI Builder FileUtils.java needs to be enabled explicitly.

berryma4 committed:
Merge pull request #2352 from css-iter/issue_2351

With the new JDOM 1.1.3 in maven-osgi-bundles the XInclude feature in…

kasemir committed:
ADL converter: Remove <wuid> from output

#2353

kasemir committed:
ADL Converter: Use actionbutton or menubutton for related displays

.. depending on the number of displays
#2353

Commit:
Reverted changes from the commit 149bc067. Added listeners for the
scrolling change.

Scrolling property listeners and fire events were (internal and
external) added to the toolbar and trace.

berryma4 committed:
remove deploy from travis, move to product repo

berryma4 committed:
travis: stdout issue from travis-ci/issues/4704

berryma4 committed:
travis: removing comment

berryma4 committed:
Merge pull request #2357 from ControlSystemStudio/issue_2355_travis

remove deploy from travis, move to product repo

berryma4 committed:
Merge pull request #2354 from ControlSystemStudio/MEDM_convert_2353

ADL converter: Remove <wuid> from output

Commit:
Fix for XInclude -- resolving entities with workspace relative path

berryma4 committed:
Revert "remove deploy from travis, move to product repo"

Commit:
Option to start window in full screen mode was added.

Window opens in full screen mode, when started from OPI editor.
start_window_in_full_screen_mode, must be set to true, if we want to
enable the feature.

kasemir committed:
vtype.pc: Make compatible with JCA 2.4.1

.. which includes CAJ and no longer offers JNI

#2361

kasemir committed:
vtype.pv: Prepare for pva update

kasemir committed:
BOY: Avoid NPE in editor for LED

Fixes #2359

berryma4 committed:
Merge pull request #2362 from ControlSystemStudio/revert-2357-issue_2355_travis

Revert "remove deploy from travis, move to product repo"

berryma4 committed:
Merge pull request #2360 from css-iter/issue_2351-1

Fix for XInclude -- resolving entities with workspace relative path

berryma4 committed:
Merge pull request #2365 from ControlSystemStudio/jca_2361

Jca 2361

berryma4 committed:
Merge pull request #2366 from ControlSystemStudio/LED_Edit_NPE_2359

BOY: Avoid NPE in editor for LED

kasemir committed:
BOY: For SWT widgets, don't overwrite the context menu ..

.. if widget itself already creates one.
Necessary to fix #2363

kasemir committed:
MEDM converter: Fix ArrayIndexOutOfBoundsException

#2368

berryma4 committed:
Merge pull request #2367 from ControlSystemStudio/BOY_SWT_widget_ctx_menu_2363

BOY: For SWT widgets, don't overwrite the context menu ..

berryma4 committed:
Merge pull request #2369 from ControlSystemStudio/MEDM_converter_2368

MEDM converter: Fix ArrayIndexOutOfBoundsException

berryma4 committed:
o.c.vtype.pv: update pvdata pvaccess names

berryma4 committed:
Merge pull request #2370 from ControlSystemStudio/issue_2341_pvaccess_name_change

o.c.vtype.pv: update pvdata pvaccess names

Commit:
opibuilder.jseditor default = false

Simple javascript editor is not default anymore, so other editors can be
set as default.

berryma4 committed:
o.c.archive.reader.ea4,o.c.alarm.beast.server,o.c.diag.epics.pvtree,o.c.saverestore.masar,o.c.scan.server: update pvdata pvaccess names

berryma4 committed:
Merge pull request #2372 from css-iter/issue_2371

opibuilder.jseditor default = false

berryma4 committed:
Merge pull request #2373 from ControlSystemStudio/issue_2341_pvaccess_name_change

o.c.archive.reader.ea4,o.c.alarm.beast.server,o.c.diag.epics.pvtree,oc.saverestore.masar,o.c.scan.server: update pvdata pvaccess names

Commit:
Preference for full screen added to the GUI. Code for resizing removed.

The code which always resized the cs-studio in case there was only one
window open in runtime mode was removed.

Full screen can be now enabled on preference page.

Commit:
Added disposing of message history GUI object.

Commit:
Control visibility of status line.

Visibility of status line in full screen or compact mode is now
separated from the setToolBar visibility method.

We can now control status line visibility in full screen and compact
mode with preferences.

berryma4 committed:
Merge pull request #2375 from css-iter/issue_2374

Issue 2374

berryma4 committed:
Merge pull request #2380 from css-iter/issue_2379

Issue 2379

Commit:
Reverted the disposing of widget.

The disposing was causing the widget disposed error and caused context
menu not working.

smarsching committed:
Fix problems with hide toolbar option.

When a plot file specified the hide toolbar option, it would still be
displayed when the plot file was loaded with the data browser OPI widget
and that widget was placed inside a container (e.g. a grouping container
or a tab).

The reason for this was that the code hiding the toolbar incorrectly
checked whether it was visible, so that it would not hide it if it was
set to visible but the container containing the data browser was not
visible yet.

berryma4 committed:
Merge pull request #2382 from smarsching/fix-hide-toolbar

Fix problems with hide toolbar option.

Commit:
Added logging of sql for message histroy at finest level.

kasemir committed:
RTPlot: Higher detail for value in annotation

#2339

berryma4 committed:
Merge pull request #2384 from ControlSystemStudio/AnnotationFormat_2339

RTPlot: Higher detail for value in annotation

kasemir committed:
RCP 'application' for edm converter

berryma4 committed:
Merge pull request #2386 from ControlSystemStudio/edl_cvt_2385

RCP 'application' for edm converter

kasemir committed:
MEDM converter: Handle "right" alignment

Was checking for "horiz.right" instead of "horiz. right".
Made it even more lenient by just checking for anything that contains
left/center/right

#2387

kasemir committed:
MEDM converter: For alarm-sensitive color, set 'OK' color to green

.. since 'OK' state uses the normal foreground color

#2387

kasemir committed:
MEDM cvt: For alarm sensitive color, disable a.s. border

kasemir committed:
MEDM cvt: Related display label when there are multiple entries

#2387

kasemir committed:
MEDM converter: Turn 'Valuator' into 'Scrollbar'

Handles direction and increment

#2387

kasemir committed:
Use 'OK' as foreground when alarm sensitive

kasemir committed:
NagTimerTest: Make more tolerant

.. to avoid failures when run in travis

berryma4 committed:
Merge pull request #2388 from ControlSystemStudio/medm_cvt_2387

Medm cvt 2387

kasemir committed:
When reading font.def, use font_default_pixels_or_points

Newer format font.def can then use "px" or "pt" on each font, but
default comes from the same preference that's used for all fonts.

Commit:
Added showing of status line to the preference pages

kasemir committed:
Alarm test: More tolerant timing for Travis

Commit:
Fixup for starting in full screen mode.

Opening in full screen mode works also when there is more than one tab open.

kasemir committed:
Fonts: Explicit "pt" overrides the default

.. similar to "px"

berryma4 committed:
Merge pull request #2390 from ControlSystemStudio/font_def_use_default_2389

When reading font.def, use font_default_pixels_or_points

kasemir committed:
MEDM converter: Create linking cont. w/o border, no resize

#2387

berryma4 committed:
Merge pull request #2392 from ControlSystemStudio/medm_cvt2_2387

MEDM converter: Create linking cont. w/o border, no resize

kasemir committed:
BOY: Omit qualifier in XML file <boy_version>

#2387

berryma4 committed:
Merge pull request #2393 from ControlSystemStudio/boy_xml_version_2387

BOY: Omit qualifier in XML file <boy_version>

Commit:
Reverted also changes to SQL for message history.

Changes to SQL to get max number messages could cause bad performance
-- getting data form SQL. This limits number of messages to max
properties.

Commit:
Reverted max messages from msghistory back to max. properties

berryma4 committed:
Merge pull request #2395 from dls-controls/2334-cursors

Improve cursor behaviour in BOY

Commit:
Fixes/robustness for restoring memento in msghistory

Commit:
Fix for detail from message history widget

kasemir committed:
MEDM Converter: Handle 'direction' for Byte Widget

#2387

kasemir committed:
MEDM Converter: Patch Visibility rules that end as "pv0====7"

#2387

kasemir committed:
MEDM converter: Guard against huge x, y, width, height values


.. which would otherwise confuse the scroll bars, or generally result in
a display that cannot be drawn on the screen.

#2387

berryma4 committed:
Merge pull request #2398 from ControlSystemStudio/medm_convert_2387

Medm convert 2387

Commit:
Fixed trailing whitespaces.

kasemir committed:
Data Browser2: Resolve macros in title 
 
 #1841

berryma4 committed:
Merge pull request #2399 from ControlSystemStudio/db2_title_macro

 Data Browser2: Resolve macros in title

berryma4 committed:
Merge pull request #2403 from css-iter/Additional_4k_icons

Additional @2x icons

berryma4 committed:
Merge pull request #2404 from css-iter/XY_graph_2133

XY graph toolbar scrolling listener added

berryma4 committed:
Merge pull request #2407 from css-iter/full_screen_fixes

Full screen fixes

Commit:
Merge branch community master into messageHistoryFixes

Commit:
Fixup for bad merge

berryma4 committed:
Merge pull request #2406 from css-iter/messageHistoryFixes

Message history fixes
